src/omnisharp/engines/LspEngine.ts

```diff
@@ -0,0 +1,330 @@
+import { EventEmitter } from 'events';
+import { LaunchInfo } from '../OmnisharpManager';
+import { Options } from '../options';
+import { setTimeout } from 'timers';
+import * as ObservableEvents from '../loggingEvents';
+import { EventStream } from '../../EventStream';
+import CompositeDisposable from '../../CompositeDisposable';
+import Disposable from '../../Disposable';
+import {
+    ServerOptions,
+    LanguageClientOptions,
+    LanguageClient,
+    Trace,
+    StaticFeature,
+    RequestType0,
+    DynamicFeature,
+    CompletionRequest,
+    HoverRequest,
+    SignatureHelpRequest,
+    DefinitionRequest,
+    ReferencesRequest,
+    DocumentHighlightRequest,
+    DocumentSymbolRequest,
+    WorkspaceSymbolRequest,
+    CodeLensRequest,
+    DocumentFormattingRequest,
+    DocumentRangeFormattingRequest,
+    DocumentOnTypeFormattingRequest,
+    RenameRequest,
+    DocumentLinkRequest,
+} from 'vscode-languageclient';
+import {
+    ExtensionContext,
+    CancellationTokenSource,
+    OutputChannel,
+} from 'vscode';
+import { ColorProviderFeature } from 'vscode-languageclient/lib/colorProvider';
+import { FoldingRangeFeature } from 'vscode-languageclient/lib/foldingRange';
+import { WorkspaceFoldersFeature } from 'vscode-languageclient/lib/workspaceFolders';
+import { ImplementationFeature } from 'vscode-languageclient/lib/implementation';
+import { DeclarationFeature } from 'vscode-languageclient/lib/declaration';
+import { SelectionRangeFeature } from 'vscode-languageclient/lib/selectionRange';
+import { TypeDefinitionFeature } from 'vscode-languageclient/lib/typeDefinition';
+import {
+    ProtocolNotificationType,
+    ProtocolNotificationType0,
+    ProtocolRequestType0,
+    ProtocolRequestType,
+} from 'vscode-languageserver-protocol/lib/messages';
+import { LanguageMiddlewareFeature } from '../LanguageMiddlewareFeature';
+import { Events } from '../server';
+import { IEngine } from './IEngine';
+
+export class LspEngine implements IEngine {
+    client: LanguageClient;
+    constructor(
+        private eventBus: EventEmitter,
+        private eventStream: EventStream,
+        private context: ExtensionContext,
+        private outputChannel: OutputChannel,
+        private disposables: CompositeDisposable,
+        private languageMiddlewareFeature: LanguageMiddlewareFeature
+    ) {}
+    public async start(
+        cwd: string,
+        args: string[],
+        launchTarget: LaunchTarget,
+        launchInfo: LaunchInfo,
+        options: Options
+    ): Promise<void> {
+        let serverOptions: ServerOptions = {
+            run: {
+                command: launchInfo.MonoLaunchPath ?? launchInfo.LaunchPath,
+                args: ['-lsp'].concat(args),
+                options: {
+                    cwd,
+                },
+            },
+            debug: {
+                command: launchInfo.MonoLaunchPath ?? launchInfo.LaunchPath,
+                args: [/*"-d", */ '-lsp'].concat(args),
```

Comment:
why is debug flag disabled in debug mode?

Reply:
I'm sure I was just testing the extension and didn't want the server to wait for a debugger to attach, I'll fix that.